core/src/com/mygdx/game/models/IntermediateBoard.java

```diff
@@ -0,0 +1,137 @@
+package com.mygdx.game.models;
+
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+
+public class IntermediateBoard {
+    private HexBoard<Terrain> terrain;
```

Comment:
We shouldn't have Terrain here. We should apply terrain like actions to the intermediatae baord.